File: todo-backend/src/main/java/bd/todo/service/TodoService.java
package bd.todo.service;

import java.util.List;

import bd.todo.model.Todo;

public interface TodoService {
	public List<Todo> getTodos();

	public List<Todo> getTodosByStatus(boolean done);

	public Todo getTodoById(Long id);

	public Todo addTodo(Todo todo);

	public Todo updateTodo(Long id, Todo todo);

	public void removeTodo(Long id);
}

File: todo-frontend/src/services/TodoService.js
import axios from 'axios';
import { API_BASE_URL } from '../constants/config';

const TODO_API_GENERAL_TODOS_URL = API_BASE_URL + "/todos";
const TODO_API_STATUS_TODOS_URL = API_BASE_URL + "/todosByStatus";

class TodoService {

    getTodos(userId) {
        return axios.get(TODO_API_GENERAL_TODOS_URL + "/" + userId);
    }

    getTodosByStatus(userId, done) {
        return axios.get(TODO_API_STATUS_TODOS_URL + "/" + userId, {
            params: {
                done: done
            }
        }
        );
    }

    saveTodo(todo) {
        if (todo.id) {
            let url = TODO_API_GENERAL_TODOS_URL + "/" + todo.id;
            return axios.put(url, todo);
        }
        else {
            return axios.post(TODO_API_GENERAL_TODOS_URL, todo);
        }
    }

    removeTodo(id) {
        let url = TODO_API_GENERAL_TODOS_URL + "/" + id;
        return axios.delete(url);
    }
}
export default new TodoService();